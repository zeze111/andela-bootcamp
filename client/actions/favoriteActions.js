import axios from 'axios';
import { FAVORITE_RECIPE, GET_FAVORITE_RECIPE, DELETE_FAVORITE } from './types';

/**
 * @export {function}
 * @param {any} recipeId
 * @returns {object} any
 */
export function favoriteRecipe(recipeId) {
  return dispatch => axios.post(`/api/v1/recipes/${recipeId}/favorite`)
    .then((response) => {
      dispatch({
        type: FAVORITE_RECIPE,
        payload: response.data,
      });
    });
}

/**
 *  @export {function}
 * @param {any} userId
 * @returns {object} any
 */
export function getFavoriteRecipes(userId) {
  return dispatch => axios.get(`/api/v1/user/${userId}/favorites`)
    .then((response) => {
      dispatch({
        type: GET_FAVORITE_RECIPE,
        payload: response.data,
      });
    });
}

/**
 *  @export {function}
 * @param {any} recipeId
 * @returns {object} any
 */
export function deleteFavorite(recipeId) {
  return dispatch => axios.delete(`/api/v1/favorites/${recipeId}/recipe`)
    .then((response) => {
      response.data.recipeId = recipeId;
      dispatch({
        type: DELETE_FAVORITE,
        payload: response.data,
      });
    });
}
