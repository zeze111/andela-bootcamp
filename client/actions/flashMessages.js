import { ADD_FLASH_MEESAGE } from './types'

export function addFlashMessages(message) {
  return (
    type: ADD_FLASH_MEESAGE,
    message
  );
}
