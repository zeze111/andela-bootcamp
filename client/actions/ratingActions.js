import axios from 'axios';
import {
  UPVOTE_RECIPE,
  DOWNVOTE_RECIPE,
  GET_DOWNVOTES,
  GET_UPVOTES,
  UPVOTE_RECIPE_FAILURE,
  DOWNVOTE_RECIPE_FAILURE,
  GET_UPVOTES_FAILURE,
  GET_DOWNVOTES_FAILURE
} from './types';
/** makes api call to get all upvotes for a recipe
 *
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @returns {object} any
 */
export const getUpvotes = recipeId =>
  dispatch => axios.get(`/api/v1/recipes/${recipeId}/upvotes`)
    .then((response) => {
      dispatch({
        type: GET_UPVOTES,
        payload: response.data,
      });
    })
    .catch((error) => {
      dispatch({
        type: GET_UPVOTES_FAILURE,
        payload: error.response.data,
      });
    });

/** makes api call to get all downvotes for a recipe
 *
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @returns {object} any
 */
export const getDownvotes = recipeId =>
  dispatch => axios.get(`/api/v1/recipes/${recipeId}/downvotes`)
    .then((response) => {
      dispatch({
        type: GET_DOWNVOTES,
        payload: response.data,
      });
    })
    .catch((error) => {
      dispatch({
        type: GET_DOWNVOTES_FAILURE,
        payload: error.response.data,
      });
    });

/** makes api call to upvote a recipe
 *
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @returns {object} any
 */
export const upvoteRecipe = recipeId =>
  dispatch => axios.post(`/api/v1/recipes/${recipeId}/upvote`)
    .then((response) => {
      dispatch(getUpvotes(recipeId));
      dispatch(getDownvotes(recipeId));
      dispatch({
        type: UPVOTE_RECIPE,
        payload: response.data,
      });
    })
    .catch((error) => {
      const toastContent = $(`<span>${error.response.data.message}</span>`);
      Materialize.toast(toastContent, 3000, 'red');
      dispatch({
        type: UPVOTE_RECIPE_FAILURE,
        payload: error.response.data,
      });
    });

/** makes api call to downvote a recipe
 *
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @returns {object} any
 */
export const downvoteRecipe = recipeId =>
  dispatch => axios.post(`/api/v1/recipes/${recipeId}/downvote`)
    .then((response) => {
      dispatch(getUpvotes(recipeId));
      dispatch(getDownvotes(recipeId));
      dispatch({
        type: DOWNVOTE_RECIPE,
        payload: response.data,
      });
    })
    .catch((error) => {
      const toastContent = $(`<span>${error.response.data.message}</span>`);
      Materialize.toast(toastContent, 3000, 'red');
      dispatch({
        type: DOWNVOTE_RECIPE_FAILURE,
        payload: error.response.data,
      });
    });
