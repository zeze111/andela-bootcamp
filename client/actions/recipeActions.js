import axios from 'axios';

import setAuthorizationToken from '../utils/setAuthorizationToken';
import uploadImageToCloud from '../utils/image';
import {
  CREATE_RECIPE,
  GET_USER_RECIPES,
  GET_ALL_RECIPES,
  DELETE_RECIPE,
  UPDATE_RECIPE,
  GET_RECIPE,
  GET_RECIPES_CATEGORY,
  SEARCH_RECIPE,
  GET_PAGED_RECIPES,
  MOST_UPVOTED_RECIPES,
  GET_ALL_RECIPES_FAILURE,
  GET_PAGED_RECIPES_FAILURE,
  GET_USER_RECIPES_FAILURE,
  GET_RECIPE_FAILURE,
  UPDATE_RECIPE_FAILURE,
  GET_RECIPES_CATEGORY_FAILURE,
  SEARCH_RECIPE_FAILURE,
  MOST_UPVOTED_RECIPES_FAILURE,
  DELETE_RECIPE_FAILURE
} from './types';

/** makes api call to add a new recipe
 *
 * @export {function}
 *
 * @param {object} recipeData form data
 *
 * @param {string} imageUrl
 *
 * @returns {object} any
 */
export function addRecipe(recipeData, imageUrl) {
  return (dispatch) => {
    setAuthorizationToken(window.localStorage.jwtToken);
    const {
      name, preparationTime, description, type, ingredients, instructions,
    } = recipeData;
    const recipe = {
      name,
      preparationTime,
      description,
      type,
      ingredients,
      instructions,
      image: imageUrl,
    };

    return axios.post('/api/v1/recipes', recipe)
      .then((response) => {
        dispatch({
          type: CREATE_RECIPE,
          payload: response.data,
        });
      });
  };
}

/** uploads recipe image to cloudinary and calls addRecipe function
 *
 * @export {function}
 *
 * @param {object} recipeData form data
 *
 * @returns {object} any
 */
export function addRecipeRequest(recipeData) {
  return (dispatch) => {
    const cloudUrl = 'http://res.cloudinary.com/zeze-andela/image/upload/v1513857342/noimg_mhvbu1.png';
    if (recipeData.imageFile.name) {
      return uploadImageToCloud(recipeData.imageFile)
        .then((response) => {
          const imageUrl = response.data.secure_url;
          dispatch(addRecipe(recipeData, imageUrl));
        })
        .catch(error => error);
    }
    return dispatch(addRecipe(recipeData, cloudUrl));
  };
}

/** makes api call to get all recipes from the database
 *
 * @export {function}
 *
 * @returns {object} any
 */
export function getAllRecipes() {
  return dispatch => axios.get('/api/v1/recipes')
    .then((response) => {
      dispatch({
        type: GET_ALL_RECIPES,
        payload: response.data,
      });
    })
    .catch((error) => {
      dispatch({
        type: GET_ALL_RECIPES_FAILURE,
        payload: error.response.data,
      });
    });
}

/** makes makes api call to get all recipes with pagination from the database
 *
 * @export {function}
 *
 * @param {number} limit
 *
 * @param {number} offset
 *
 * @returns {object} any
 */
export function getPaginatedRecipes(limit, offset) {
  return dispatch =>
    axios.get(`/api/v1/recipes/?limit=${limit}&offset=${offset}`)
      .then((response) => {
        dispatch({
          type: GET_PAGED_RECIPES,
          payload: response.data,
        });
      })
      .catch((error) => {
        dispatch({
          type: GET_PAGED_RECIPES_FAILURE,
          payload: error.response.data,
        });
      });
}

/** makes api call to get all recipes submitted by a user
 *
 * @param {number} limit
 *
 * @param {number} offset
 *
 * @export {function}
 *
 * @returns {object} any
 */
export function getUserRecipes(limit, offset) {
  return dispatch =>
    axios.get(`/api/v1/user/recipes?limit=${limit}&offset=${offset}`)
      .then((response) => {
        dispatch({
          type: GET_USER_RECIPES,
          payload: response.data,
        });
      })
      .catch((error) => {
        dispatch({
          type: GET_USER_RECIPES_FAILURE,
          payload: error.response.data,
        });
      });
}

/** makes api call to get a single recipe
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @returns {object} any
 */
export function getARecipe(recipeId) {
  return dispatch => axios.get(`/api/v1/recipes/${recipeId}`)
    .then((response) => {
      dispatch({
        type: GET_RECIPE,
        payload: response.data,
      });
    })
    .catch((error) => {
      dispatch({
        type: GET_RECIPE_FAILURE,
        payload: error.response.data,
      });
    });
}

/** makes api call to update a recipe
 *
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @param {object} recipeData form data
 *
 * @returns {object} any
 */
export function updateRecipe(recipeId, recipeData) {
  return dispatch => axios.put(`/api/v1/recipes/${recipeId}`, recipeData)
    .then((response) => {
      dispatch({
        type: UPDATE_RECIPE,
        payload: response.data,
      });
    })
    .catch((error) => {
      dispatch({
        type: UPDATE_RECIPE_FAILURE,
        payload: error.response.data,
      });
    });
}

/** makes api call to delete a user's recipe
 *
 * @export {function}
 *
 * @param {number} recipeId
 *
 * @returns {object} any
 */
export function deleteRecipe(recipeId) {
  return dispatch => axios.delete(`/api/v1/recipes/${recipeId}`)
    .then(() => {
      dispatch({
        type: DELETE_RECIPE,
        payload: recipeId,
      });
    })
    .catch((error) => {
      dispatch({
        type: DELETE_RECIPE_FAILURE,
        payload: error.response.data,
      });
    });
}

/** makes api call to get all recipes matching a category type
 *
 * @export {function}
 *
 * @param {string} type category type
 *
 * @param {number} limit
 *
 * @param {number} offset
 *
 * @returns {object} any
 */
export function getRecipeCategory(type, limit, offset) {
  return dispatch =>
    axios.get(`/api/v1/recipes/categories/${type}?limit=${limit}&offset=${offset}`)
      .then((response) => {
        dispatch({
          type: GET_RECIPES_CATEGORY,
          payload: response.data,
        });
      })
      .catch((error) => {
        dispatch({
          type: GET_RECIPES_CATEGORY_FAILURE,
          payload: error.response.data,
        });
      });
}

/** makes api call to search for a recipe match
 *
 * @export {function}
 *
 * @param {string} word search word
 *
 * @param {number} limit
 *
 * @param {number} offset
 *
 * @returns {object} any
 */
export function searchRecipe(word, limit, offset) {
  return dispatch =>
    axios.get(`/api/v1/recipes/search/${word}?limit=${limit}&offset=${offset}`)
      .then((response) => {
        dispatch({
          type: SEARCH_RECIPE,
          payload: response.data,
        });
      })
      .catch((error) => {
        dispatch({
          type: SEARCH_RECIPE_FAILURE,
          payload: error.response.data,
        });
      });
}

/** makes api call to get most upvoted recipes
 *
 * @export {function}
 *
 * @returns {object} any
 */
export function getMostUpvotedRecipe() {
  return dispatch => axios.get('/api/v1/recipes/?sort=upvotes&order=des')
    .then((response) => {
      dispatch({
        type: MOST_UPVOTED_RECIPES,
        payload: response.data,
      });
    })
    .catch((error) => {
      dispatch({
        type: MOST_UPVOTED_RECIPES_FAILURE,
        payload: error.response.data,
      });
    });
}
