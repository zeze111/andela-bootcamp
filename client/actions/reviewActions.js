import axios from 'axios';
import { REVIEW_RECIPE, GET_REVIEWS, DELETE_REVIEW } from './types';

/**
 * @export {function}
 * @param {any} recipeId
 * @param {any} review
 * @returns {object} any
 */
export function reviewRecipe(recipeId, review) {
  return dispatch => axios.post(`/api/v1/recipes/${recipeId}/review`, review)
    .then((response) => {
      dispatch({
        type: REVIEW_RECIPE,
        payload: response.data
      });
    })
    .catch(error => error);
}

/**
 * @export {function}
 * @param {any} recipeId
 * @returns {object} any
 */
export function getReviews(recipeId) {
  return dispatch => axios.get(`/api/v1/recipes/${recipeId}/reviews`)
    .then((response) => {
      dispatch({
        type: GET_REVIEWS,
        payload: response.data,
      });
    });
}

/**
 * @export {function}
 * @param {any} reviewId
 * @returns {object} any
 */
export function deleteReview(reviewId) {
  return dispatch => axios.delete(`/api/v1/recipes/${reviewId}/review`)
    .then((response) => {
      Materialize.toast(`<span> ${response.data.message}</span>`, 2000);

      dispatch({
        type: DELETE_REVIEW,
        payload: reviewId,
      });
    });
}
