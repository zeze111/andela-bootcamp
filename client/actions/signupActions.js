import axios from 'axios';
import jwt from 'jsonwebtoken';
import setAuthorizationToken from '../utils/setAuthorizationToken';
import { SET_CURRENT_USER } from './types';

export function setCurrentUser(user) {
  return {
    type: SET_CURRENT_USER,
    user,
  };
}

export function userSignupRequest(userData) {
  return dispatch => axios.post('/api/v1/users/signup', userData)
    .then((res) => {
      const { token, user } = res.data;
      const { id, email, firstName } = user;
      localStorage.setItem('jwtToken', token);
      localStorage.setItem('user', JSON.stringify({ id, email, firstName }));
      setAuthorizationToken(token);
      dispatch(setCurrentUser({ id, email, firstName }));
    });
}
