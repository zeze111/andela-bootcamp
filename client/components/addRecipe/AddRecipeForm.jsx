import React from 'react';
import PropTypes from 'prop-types';
import Validator from 'validatorjs';
import { pick } from 'lodash';
import { Input, Row } from 'react-materialize';
import { Redirect } from 'react-router-dom';
import validations from '../../../Server/shared/validations';
import { TextFieldGroup, TextFieldGroup2 } from '../common/TextFieldGroup';
import PreLoader from '../common/PreLoader';

/** Form to allow users create s recipe
 *
 * @class AddRecipeForm
 *
 * @extends {React.Component}
 */
class AddRecipeForm extends React.Component {
  /**
   * @description Constructor Function
   *
   * @param {object} props
   *
   * @memberof Home
   *
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      name: '',
      preparationTime: '',
      description: '',
      type: '',
      ingredients: '',
      instructions: '',
      imageSrc: '/images/noimg.png',
      imageFile: '',
      errors: {},
      isLoading: false
    };
  }

  /** css initialisation for when a component is mounted
   *
   * @memberof Home
   *
   * @return {void}
   */
  componentDidMount() {
    $(this.refs.selectType).material_select(this.onSelectChange.bind(this));
  }

  /** sets state when form input is changed
   *
   * @param {object} event
   *
   * @memberof Home
   *
   * @return {void}
   */
  onChange = (event) => {
    this.setState({ [event.target.name]: event.target.value });
  }

  /** sets state when a type field is selected
   *
   * @param {object} event
   *
   * @memberof Home
   *
   * @return {void}
   */
  onSelectChange = (event) => {
    const { value } = event.target;
    this.setState({
      type: value
    });
  }

  /** calls action to add user's recipe to the app
   *
   * @param {object} event
   *
   * @memberof Home
   *
   * @return {void}
   */
  onSubmit = (event) => {
    event.preventDefault();

    const recipe = pick(this.state, ['name', 'preparationTime', 'description', 'type',
      'ingredients', 'instructions', 'imageFile']);

    if (this.isValid()) {
      this.setState({ errors: {}, isLoading: true });
      this.props.addRecipeRequest(recipe)
        .then(() => this.setState({ redirect: true }))
        .catch(error => this.setState({
          errors: error.response.data, isLoading: false
        }));
    }
  }


  /** ets state of imageFile and changes recipe image display
   *
   * @param {object} event
   *
   * @memberof Home
   *
   * @return {void}
   */
  uploadImage = (event) => {
    if (event.target.files && event.target.files[0]) {
      this.setState({ imageFile: event.target.files[0] });
      const reader = new FileReader();
      reader.onload = (readerEvent) => {
        this.setState({ imageSrc: readerEvent.target.result });
      };
      reader.readAsDataURL(event.target.files[0]);
    } else {
      this.setState({ imageSrc: '/images/noimg.png', imageFile: '' });
    }
  }

  /** checks if form validation passes or fails
   *
   * @memberof Home
   *
   * @return {boolean} validator
   */
  isValid() {
    const validator = new Validator(this.state, validations.recipeRules);
    if (validator.fails()) {
      const errors = validator.errors.all();
      this.setState({ errors });
    }

    return validator.passes();
  }

  /** html component to render
   *
   * @memberof Home
   *
   * @return {void}
   */
  render() {
    const { errors } = this.state;
    const { redirect } = this.state;
    if (redirect) {
      return <Redirect to="/user/2/recipes" />;
    }

    return (
      <div className="row">
        <div className="card-content col s10 m8 l6 push-s1 push-m2 push-l3">
          <form onSubmit={this.onSubmit}>
            <div className="row add-padding">
              <div className="file-field input-field col s12 right-align get-pic-btn" >
                <div className="image-placeholder">
                  <img
                    src={this.state.imageSrc}
                    alt=""
                    className="recipe-image"
                  />
                </div>
                <div
                  className="btn-floating
                btn-small
                waves-effect
                waves-light
                blue"
                >
                  <i className="material-icons">photo</i>
                  <input type="file" onChange={this.uploadImage} />
                </div>
              </div>
            </div>
            <div className="half-top">
              {errors &&
                <span className="red-text error-text" >
                  {errors.message}
                </span>}
              <TextFieldGroup
                label="Recipe Name"
                value={this.state.name}
                onChange={this.onChange}
                id="name"
                type="text"
                name="name"
                error={errors.name}
              />
              <TextFieldGroup
                label="Recipe Time"
                value={this.state.preparationTime}
                onChange={this.onChange}
                id="time"
                type="text"
                name="preparationTime"
                error={errors.preparationTime}
              />
              <TextFieldGroup
                label="Description"
                value={this.state.description}
                onChange={this.onChange}
                id="desc"
                type="text"
                name="description"
                error={errors.description}
              />
              <Row>
                <Input
                  s={12}
                  type="select"
                  label="Recipe Type"
                  name="type"
                  defaultValue="0"
                  onChange={this.onSelectChange}
                >
                  <option value="0" disabled> Choose your option </option>
                  <option value="Appetizer">Appetizer</option>
                  <option value="Main" >Main</option>
                  <option value="Dessert" >Dessert</option>
                  <option value="Drink" >Drink</option>
                </Input>
                {errors &&
                  <span className="help-block red-text error-text2">
                    {errors.type}
                  </span>}
              </Row>
              <TextFieldGroup2
                label="Ingredients"
                value={this.state.ingredients}
                onChange={this.onChange}
                id="ingred"
                name="ingredients"
                error={errors.ingredients}
              />
              <TextFieldGroup2
                label="Instructions"
                value={this.state.instructions}
                onChange={this.onChange}
                id="instruct"
                name="instructions"
                error={errors.instructions}
              />
              <div className="right-align">
                {
                  (this.state.isLoading) &&
                  <div className="center-align loader-style">
                    <PreLoader />
                  </div>
                }
                <input
                  className="btn grey"
                  type="submit"
                  value="Submit"
                  disabled={this.state.isLoading}
                />
              </div> <br />
            </div>
          </form>
        </div>
      </div>
    );
  }
}

AddRecipeForm.propTypes = {
  addRecipeRequest: PropTypes.func.isRequired,
};

export default AddRecipeForm;
