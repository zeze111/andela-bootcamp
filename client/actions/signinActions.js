import axios from 'axios';

import setAuthorizationToken from '../utils/setAuthorizationToken';
import { setCurrentUser } from './signupActions';


/** signs user out of the app
 *
 * @export {function}
 *
 * @returns {object} any
 */
export function signout() {
  return (dispatch) => {
    localStorage.removeItem('jwtToken');
    setAuthorizationToken(false);
    dispatch(setCurrentUser(undefined, undefined));
  };
}

/** makes api call to sign user in and dispatch the response
 * 
 * @export {function}
 *
 * @param {object} userData form data
 *
 * @returns {object} any
 */
export function userSigninRequest(userData) {
  return dispatch => axios.post('/api/v1/users/signin', userData)
    .then((response) => {
      const { token, user } = response.data;
      const { id, email, firstName } = user;
      localStorage.setItem('jwtToken', token);
      setAuthorizationToken(token);
      dispatch(setCurrentUser({ id, email, firstName }));
    });
}
