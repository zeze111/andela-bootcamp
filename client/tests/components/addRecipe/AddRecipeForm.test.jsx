import React from 'react';

import AddRecipeForm from '../../../components/addRecipe/AddRecipeForm';
import store from '../../../store';
import { addRecipeRequest } from '../../../actions/recipeActions';

let props = {
  addRecipeRequest: jest.fn(() => Promise.resolve())
};

const setup = () => {
  const wrapper = shallow(<AddRecipeForm {...props} store={store} />);
  return { wrapper }
}

describe('onChange', () => {
  it('should change event', () => {
    const event = {
      target: { name: 'name', value: 'Rice and stew' },
    };
    const { wrapper } = setup();
    const recipeName = wrapper.find('#name');
    recipeName.simulate('change', event);
    expect(wrapper.instance().state.name).toBe('Rice and stew');
  });
});

describe('onSelectChange', () => {
  it('should change event for type select field', () => {
    const event = {
      target: { value: 'Main' },
    };
    const { wrapper } = setup();
    const type = wrapper.find('#type');
    type.simulate('change', event);
    expect(wrapper.instance().state.type).toBe('Main');
  });
});

describe('onSubmit', () => {
  it('be called when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };
    wrapper.setState({
      name: 'Rice and stew',
      ingredients: 'rice, tomatoes, pepper, onions',
      instructions: 'Boil the rice',
      imageFile: { name: 'pic.jpg' },
      preparationTime: '12 minutes',
      type: 'Main',
      errors: {},
      isLoading: false
    });
    wrapper.instance().onSubmit(event);
    expect(props.addRecipeRequest.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return validation error when form is submitted', () => {
    const { wrapper } = setup();
    const event = {
      preventDefault: jest.fn(),
    };

    wrapper.setState({
      name: 'Rice and stew',
      ingredients: 'rice, tomatoes, pepper, onions',
      instructions: 'Boil the rice',
      imageFile: { name: 'pic.jpg' },
      preparationTime: '12 minutes'
    });
    wrapper.instance().onSubmit(event);
    expect(props.addRecipeRequest.mock.calls.length).toEqual(1);
  });
});

describe('onSubmit', () => {
  it('return an error when form is submitted', () => {
    const event = {
      preventDefault: jest.fn(),
    };
    props.addRecipeRequest = jest.fn(() => Promise.reject());
    const wrapper = shallow(<AddRecipeForm {...props} store={store} />);

    wrapper.setState({
      name: 'Rice and stew',
      ingredients: 'rice, tomatoes, pepper, onions',
      instructions: 'Boil the rice',
      imageFile: { name: 'pic.jpg' },
      preparationTime: '12 minutes',
      type: 'Main',
      errors: {},
      isLoading: false
    });
    wrapper.instance().onSubmit(event);
    expect(props.addRecipeRequest.mock.calls.length).toEqual(1);
  });
});


// describe('upLoadImage', () => {
//   it('be called when recipe image is selected', () => {
//     const { wrapper } = setup();
//     wrapper.setState({
//       imageSrc: 'pic.jpg',
//     });
//     const file = {
//       target: {
//         files: ['one', 'two']
//       }
//     };

//     reader = {
//       onload: jest.fn()
//     }
//     wrapper.instance().uploadImage(reader);
//     expect(wrapper.instance().state.imageSrc).toBe('pic.jpg');
//   });
// });

describe('Test for Add Recipe Form', () => {
  it('should render component', () => {
    const { wrapper } = setup();
    expect(wrapper).toMatchSnapshot()
  });
});
