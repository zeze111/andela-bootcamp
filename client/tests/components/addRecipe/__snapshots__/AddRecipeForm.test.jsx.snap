// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for Add Recipe Form should render component 1`] = `
<div
  className="row"
>
  <div
    className="card-content col s10 m8 l6 push-s1 push-m2 push-l3"
  >
    <form
      onSubmit={[Function]}
    >
      <div
        className="row add-padding"
      >
        <div
          className="file-field input-field col s12 right-align get-pic-btn"
        >
          <div
            className="image-placeholder"
          >
            <img
              alt=""
              className="recipe-image"
              src="/images/noimg.png"
            />
          </div>
          <div
            className="btn-floating btn-small waves-effect waves-light orange"
          >
            <i
              className="material-icons"
            >
              photo
            </i>
            <input
              onChange={[Function]}
              type="file"
            />
          </div>
        </div>
      </div>
      <div
        className="half-top"
      >
        <span
          className="red-text error-text"
        />
        <TextFieldGroup
          active=""
          error={Array []}
          id="name"
          label="Recipe Name"
          name="name"
          onChange={[Function]}
          placeholder=""
          req="*"
          type="text"
          value=""
        />
        <TextFieldGroup
          active="active"
          error={Array []}
          id="time"
          label="Preparation Time"
          name="preparationTime"
          onChange={[Function]}
          placeholder="e.g. 1 hour / 2hrs 30mins"
          req="*"
          type="text"
          value=""
        />
        <TextFieldGroup
          active="active"
          error={Array []}
          id="desc"
          label="Description"
          name="description"
          onChange={[Function]}
          placeholder="e.g. Great for casual dinner parties"
          req=""
          type="text"
          value=""
        />
        <Row>
          <Input
            defaultValue="0"
            id="type"
            label="Recipe Type"
            name="type"
            onChange={[Function]}
            s={12}
            type="select"
          >
            <option
              disabled={true}
              value="0"
            >
               Choose your option 
            </option>
            <option
              value="Appetizer"
            >
              Appetizer
            </option>
            <option
              value="Main"
            >
              Main
            </option>
            <option
              value="Dessert"
            >
              Dessert
            </option>
            <option
              value="Drinks"
            >
              Drinks
            </option>
          </Input>
          <span
            className="help-block red-text error-text2"
          />
        </Row>
        <TextFieldGroup2
          error={Array []}
          id="ingred"
          label="Ingredients"
          name="ingredients"
          onChange={[Function]}
          placeholder="e.g. 2 cups of water, 1 tsp salt, tomatoes"
          req="*"
          value=""
        />
        <TextFieldGroup2
          error={Array []}
          id="instruct"
          label="Instructions"
          name="instructions"
          onChange={[Function]}
          placeholder="e.g. Preheat oven at 120 celcius. boil egg."
          req="*"
          value=""
        />
        <span
          className="blue-text"
        >
          Seperate instructions by full stop (.)
        </span>
        <div
          className="right-align"
        >
          <input
            className="btn grey"
            disabled={false}
            type="submit"
            value="Submit"
          />
        </div>
         
        <br />
      </div>
    </form>
  </div>
</div>
`;
